torchtitan/models/deepseek_v3/infra/parallelize.py

```diff
@@ -19,5 +19,32 @@ def parallelize_deepseekv3(
     parallel_dims: ParallelDims,
     job_config: JobConfig,
 ):
-    # TODO: Add support for parallelizing the model, this is a placeholder function for now
+    if job_config.activation_checkpoint.mode != "none":
+        apply_ac(model, job_config.activation_checkpoint)
```

Comment:
My understanding is that for SAC we are counting the number of matmuls occuring during forward, then selectively saving every say, N matmuls.

MoE might affect this in two ways:
1. matmul imbalances (gating/routing computation is lightweight, while expert MM is heavy)
2. Not sure how this interacts with expert parallel is across multiple ranks?

I'm not sure if we cover this in Llama4, any ideas @tianyu-l? Anyways, if SAC isn't covered i dont think its that high pri but maybe just add a comment.